### PR TITLE
Add Visualization utilities article

### LINKED ISSUE #113: Add vignette on 'Visualization utilities'
As a port of [Visualization Utilities](https://pytorch.org/vision/stable/auto_examples/others/plot_visualization_utils.html#sphx-glr-auto-examples-others-plot-visualization-utils-py)

### LINKED ISSUE #383: Add vignette on 'Visualization utilities' demonstrating object detection model visual results and segmentation model visual results
Goal is to have a vignette with the same stories as in  [Visualization Utilities](https://pytorch.org/vision/stable/auto_examples/others/plot_visualization_utils.html#sphx-glr-auto-examples-others-plot-visualization-utils-py)
 - relying on RF-DETR model for object detection
 - having the vignette published as a web-site article.

This superseeds and replace #113

### PR DESCRIPTION
**Pull request by srishtiii28**
Fix #383.

Adds a Visualization utilities article mirroring PyTorch's tutorial: `vision_make_grid()`, `draw_bounding_boxes()` with RF-DETR, `draw_segmentation_masks()` with FCN ResNet-50, and `draw_keypoints()`.

Fix #113

### CODE REVIEW

**Comment by srishtiii28** [comment]
Instance segmentation is left out here because Mask R-CNN's masks never cross the 0.5 threshold, so the figures come out blank. Should I work on fixing that? @cregouby

**Comment by cregouby** [comment]
Yes, both Mask R-CNN and Faster R-CNN  are very sensitive to score threshold and nms threshold (see examples or each function). 
Please open an issue with it, and then you may fix it.

File: vignettes/examples/visualization-utilities.Rmd [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+dog2 <- base_loader(paste0(gallery, "dog2.jpg")) %>% transform_to_tensor()
+```
+
+## A grid of images
+
+`vision_make_grid()` tiles a batch of images into a single tensor, which is a
+quick way to eyeball a batch coming out of a `dataloader()`.
+
+```{r}
+grid <- vision_make_grid(torch_stack(list(dog1, dog2)), num_rows = 2)
```
Comment (cregouby) [inline vignettes/examples/visualization-utilities.Rmd:35]: **thought** Should we wait for #394 Solution 2 available in #398 and have a nice looking `grid <- vision_make_grid(dog1, dog2, num_rows = 2)` ?

File: vignettes/examples/visualization-utilities.Rmd [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+`draw_bounding_boxes()` expects an `(N, 4)` tensor of boxes given as
+`(xmin, ymin, xmax, ymax)` in absolute pixel coordinates.
+
+```{r}
+boxes <- torch_tensor(rbind(
+  c(50, 50, 100, 200),
+  c(210, 150, 350, 430)
+), dtype = torch_float())
+
+boxed <- draw_bounding_boxes(dog1, boxes, colors = c("red", "yellow"), width = 5)
```
Comment (cregouby) [inline vignettes/examples/visualization-utilities.Rmd:52]: **suggestion** could we add labels = c("label_1", ...) to visually explain all the features of the function ?

File: vignettes/examples/visualization-utilities.Rmd [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+0.91 with normalization and 0.69 without.
+
+```{r}
+norm_mean <- c(0.485, 0.456, 0.406)
+norm_std <- c(0.229, 0.224, 0.225)
+
+batch <- torch_stack(list(
+  transform_normalize(dog1, norm_mean, norm_std),
+  transform_normalize(dog2, norm_mean, norm_std)
+))
```
Comment (cregouby) [inline vignettes/examples/visualization-utilities.Rmd:72]: **suggestion** if I'm not wrong, transform_normalize() accepts batches, so I would demonstrate it and I would apply it once after torch_strack()

File: vignettes/examples/visualization-utilities.Rmd [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+  keep <- as.logical(as.array(detection$scores > 0.5))
+  draw_bounding_boxes(
+    image,
+    detection$boxes[keep, , drop = FALSE],
+    labels = coco_classes(as.integer(detection$labels[keep])),
+    colors = "yellow",
+    width = 5,
+    font_size = 30
+  )
+}
```
Comment (cregouby) [inline vignettes/examples/visualization-utilities.Rmd:96]: **thought** I'm not in favor of defining new functions wrapping visualization utilities in an article where the intent is to demonstrate how good the visualization utilities are. That said, I've no better idea to not have a lot of boilerplate code. 
**suggestion** We should at least explain the why of this function, and what each step does. 
**suggestion** a small size, naive visualization of the direct model output (normalized image and unfiltered bbox) would allow a good justification of the why this function

File: vignettes/examples/visualization-utilities.Rmd [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+
+## Semantic segmentation masks
+
+`model_fcn_resnet50()` scores every pixel against the 21 PASCAL VOC classes. Its
+pretrained weights are evaluated at 520 pixels, so resize before normalizing,
+and keep the resized images around to draw on: masks and image must share their
+height and width.
+
+```{r}
+resized <- lapply(list(dog1, dog2), transform_resize, size = c(520, 520))
```
Comment (cregouby) [inline vignettes/examples/visualization-utilities.Rmd:113]: **todo** **simplification** **clarity** transform_resize is also accepting batches. So please demonstrate the piping of transforms after stacking the images.

**Review by cregouby — CHANGES_REQUESTED** [review]
**Praise** Very nice vignette, thank you !
**todo** see inline
